### DISCUSSION TITLE
contributing to  open issues

### DISCUSSION THREAD

**Discussion by razvangoga** [akkadotnet/akka.net#4722] [General]
Hi @Aaronontheweb, @IgorFedchenko 

I'm having some free time in the next 2 weeks (maybe around 6-8h combined) and could try to take a look at some other open issues. 
Do you guys have any one in particular in mind that may be more pressing? 
Or I can just take my pick from the good-for-first-time-contribs / up-for-grabs ones?

Best,
Razvan

**Comment by Aaronontheweb** [comment]
Hi @razvangoga 

Thank you very much for your kind offer!

One thing that immediately springs to mind that would be helpful is addressing some of the racy unit tests in our test suite: https://dev.azure.com/dotnet/Akka.NET/_test/analytics?definitionId=84&contextType=build

These create a lot of problems that disrupt our ability to merge in simple pull requests, i.e. https://github.com/akkadotnet/akka.net/pull/4719 and they undermine confidence in the test suite.

Some of these racy unit tests are the result of poor test design, others are the result of creating tight temporal coupling (i.e. we need to assert that X Akka.Streams stage completes in 100ms, but we can't actually guarantee that), and in others it might be real bugs that are causing the problem. It's difficult to find the third category of problem when the first two create so much noise.

That link I provided will give you the set of tests with the highest pass -> fail -> pass -> fail "flip rate" - if you could look at patching some of those, that would be a huge time saver for us.

Aside from that, any of the "[up for grabs](https://github.com/akkadotnet/akka.net/issues?q=is%3Aissue+is%3Aopen+label%3A%22up+for+grabs%22)" or "[help wanted](https://github.com/akkadotnet/akka.net/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)" issues would be a great start!

**Comment by IgorFedchenko** [comment]
@razvangoga Yep, I agree, fixing racy specs is extremely helpful for everybody who is ever contributing to Akka.NET.

As an addition to the build stats Aaron gave you, you can check out this issue, especially the diagram Aaron has drawn: https://github.com/akkadotnet/akka.net/issues/3786#issuecomment-560564543 . That should help you figure out what to do with those specs that do have a high flip rate.

**Comment by razvangoga** [comment]
Sure, I'll take a look and come back with any questions I (for sure )might have

**Reply by Aaronontheweb** [reply]
Let us know!
